File: 2024/java/junior/Junior4.java
import java.util.*;

public class Junior4 {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        String text = in.nextLine();
        String output = in.nextLine();

        boolean hasSilent = text.length() > output.length();
        List<Character> mismatch = new ArrayList<>();
        char replace = ' ';

        int[] wordCount = new int[26];
        int[] outputCount = new int[26];
        for (int i = 0; i < text.length(); i++) {
            wordCount[text.charAt(i) - 'a']++;
        }
        for (int i = 0; i < output.length(); i++) {
            outputCount[output.charAt(i) - 'a']++;
        }
        for (int i = 0; i < 26; i++) {
            if (wordCount[i] > outputCount[i]) {
                mismatch.add((char) ('a' + i));
            }
            if (wordCount[i] < outputCount[i]) {
                replace = (char) ('a' + i);
            }
        }
        String newText = text.replace(mismatch.get(0), replace);
        if (hasSilent) {
            newText = newText.replaceAll("" + mismatch.get(1), "");
        }
        if (newText.equals(output)) {
            System.out.println(mismatch.get(0) + " " + replace);
            if (hasSilent) {
                System.out.println(mismatch.get(1));
            } else {
                System.out.println("-");
            }
        } else {
            System.out.println(mismatch.get(1) + " " + replace);
            System.out.println(mismatch.get(0));
        }
    }
}

File: 2024/cpp/junior/junior4.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    string input, output;
    cin >> input;
    cin >> output;

    vector<char> mismatch;
    char extra = ' ';
    bool hasSilent = input.length() > output.length();
    vector<int> inputCount(26), outputCount(26);
    for (char c : input) {
        inputCount[c - 'a']++;
    }
    for (char c : output) {
        outputCount[c - 'a']++;
    }

    for (int i = 0; i < 26; i++) {
        if (inputCount[i] > outputCount[i]) {
            mismatch.push_back('a' + i);
        }
        if (inputCount[i] < outputCount[i]) {
            extra = 'a' + i;
        }
    }
    replace(input.begin(), input.end(), mismatch[0], extra);
    if (hasSilent) {
        input.erase(remove(input.begin(), input.end(), mismatch[1]), input.end());
    }
    if (input == output) {
        cout << mismatch[0] << " " << extra << endl;
        if (hasSilent) {
            cout << mismatch[1] << endl;
        } else {
            cout << "-" << endl;
        }
    } else {
        cout << mismatch[1] << " " << extra << endl;
        cout << mismatch[0] << endl;
    }
}